eql/parser.py

```diff
@@ -1023,6 +1023,19 @@ def subquery_by(self, node, num_values=None, position=None, close=None, allow_fo
             raise self._error(error_node, message, num=num_values, end=end)
 
         kwargs = {}
+
+        repeated_sequence = node["repeated_sequence"]
+
+        runs_count = 1
+        if repeated_sequence is not None:
+            runs_count = node["repeated_sequence"]["UNSIGNED_INTEGER"].value
+
+            if int(runs_count) <= 0:
+                raise self._error(repeated_sequence, "Repeated sequence runs must be greater than 0")
+
+            if allow_runs is False:
+                raise self._error(repeated_sequence, "Repeated sequences are not allowed here")
```

Comment:
can you check this one first? syntax errors should take priority.
minor language nit: In the past, I've tried not to refer to the location in the text, just in the `^^^^` so that you can still see the error message in isolation without the context and known what went wrong.

```suggestion
            if allow_runs is False:
                raise self._error(repeated_sequence, "Unsupported usage of repeated syntax", cls=EqlSyntaxError)
```